### DISCUSSION TITLE
How to Inject Interface in Aspect

### DISCUSSION THREAD

**Discussion by MajidMoghise** [pamidur/aspect-injector#211] [General]
I want to inject Interface in Aspect For example inject Ilogger or IserviceProvider interface  or Any interface in constructor 

` 
   [Aspect(Scope.Global)]
    [Injection(typeof(LogCall))]
    public class LogCall : Attribute
    {
        private readonly ILogger<LogCall> _logger;
        public LogCall(ILogger<LogCall> logger)
        {
            _logger = logger;
        }
        [Advice(Kind.Before)]
        public void LogEnter([Argument(Source.Name)] string name, [Argument(Source.Arguments)] object[] objcts)
        {
           _logger.Information($"Calling '{name}'( " + JsonConvert.SerializeObject(objcts) + " )"); 
        }
        [Advice(Kind.After)] 
        public void LogOut([Argument(Source.Name)] string name, [Argument(Source.ReturnValue)] object objcts)
        {
             _logger.Information($"Out '{name}'( " + JsonConvert.SerializeObject(objcts) + " )");  
        }
`

**Comment by pamidur** [comment]
Please see #166
